core/src/main/scala/spark/Utils.scala

```
@@ -287,29 +288,14 @@ private object Utils extends Logging {
     customHostname.getOrElse(InetAddress.getLocalHost.getHostName)
   }
 
-  /**
-   * Returns a standard ThreadFactory except all threads are daemons.
-   */
-  private def newDaemonThreadFactory: ThreadFactory = {
-    new ThreadFactory {
-      def newThread(r: Runnable): Thread = {
-        var t = Executors.defaultThreadFactory.newThread (r)
-        t.setDaemon (true)
-        return t
-      }
-    }
-  }
+  private[spark] val daemonThreadFactory: ThreadFactory =
```

@woggling added a default uncaught exception handler in #305 but this only affected worker machines.  This handler ensures that Executors die rather than hanging if they run out of memory or experience other types of failures.  I'm not sure if we want to implement the same "fail fast" behavior in the master or driver, since they might be embedded with other user code that doesn't expect that behavior.